dash-pipeline/SAI/templates/saiapi.h.j2

```diff
@@ -34,7 +34,7 @@
  */
 
 {% for table in sai_api.tables %}
-{% if table.actions | length > 1 %}
+{% if table.actions | length > 1 or ((table.actions | length == 1) and (((table.is_object == 'false') or (table['keys'] | length <= 1)) and ((table['actionParams'] | length == 0))))  %}
```

Comment:
Original condition tries to generate action-type even for sai object if it has only 1 action with 0 params. Also it skips 0 action case. If these are no longer needed / not possible, then this simpler should be fine.